pkgs/tools/security/ssh-mitm/default.nix

```
@@ -1,55 +1,43 @@
-{ lib
-, fetchFromGitHub
-, python3
+{ lib,
+fetchFromGitHub,
+python3,
+stdenv
 }:
 
-let
-  py = python3.override {
-    packageOverrides = self: super: {
-      paramiko = super.paramiko.overridePythonAttrs (oldAttrs: rec {
-        version = "3.1.0";
-        src = oldAttrs.src.override {
-          inherit version;
-          hash = "sha256-aVD6ymgZrNMhnUrmlKI8eofuONCE9wwXJLDA27i3V2k=";
-        };
-        patches = [ ];
-        propagatedBuildInputs = oldAttrs.propagatedBuildInputs ++ [ python3.pkgs.icecream ];
-      });
-    };
-  };
-in
-with py.pkgs;
-
-buildPythonApplication rec {
+python3.pkgs.buildPythonApplication rec {
   pname = "ssh-mitm";
-  version = "3.0.2";
+  version = "4.1.1";
   format = "setuptools";
```

```suggestion
  pyproject = true;
```

and then 

```
  build-system = with python3.pkgs; [
    hatchling
    hatch-requirements-txt
];
```

see https://github.com/ssh-mitm/ssh-mitm/blob/4.1.1/pyproject.toml#L1